Photo-Editor/Photo-Editor/Views/HomeScreenView.swift

```diff
@@ -8,35 +8,199 @@
 import SwiftUI
 
 struct HomeScreenView: View {
-    @State private var recentImages: [UIImage] = []
+    @Binding var recentImages: [UIImage]
+    @Binding var trashedImages: [UIImage]
     @State private var selectedImage: UIImage?
     @State private var isEditImageViewActive = false
-
+    @State private var refreshID = UUID()
+    
     var body: some View {
         NavigationStack {
             VStack {
                 ScrollView {
                     MainButton(recentImages: $recentImages)
+
                     Spacer()
-                    HomeLibrary(recentImages: recentImages, onImageTap: { image in
-                        selectedImage = image
-                        isEditImageViewActive = true
-                    })
+
+                    HomeLibrary(
+                        recentImages: recentImages,
+                        onImageTap: { image in
+                            selectedImage = image
+                            isEditImageViewActive = true
+                        },
+                        onImageDelete: { image in
+                            moveImageToTrash(image: image)
+                        }
+                    )
+                    .id(refreshID)
                 }
                 .navigationTitle("Choose an image")
                 .navigationBarTitleDisplayMode(.inline)
 
                 NavigationLink(
-                    destination: EditImageView(image: selectedImage ?? UIImage()),
```

Comment:
You should be getting a warning on the NavigationLink. You are using the deprecated version of the API. The warning message suggests what to use instead.